lessons/advanced/typespec.md

```diff
@@ -0,0 +1,197 @@
+---
+layout: page
+title: Specifications and types
+category: advanced
+order: 9
+lang: en
+---
+
+If you want to define 'interface' like in Java or Ruby, Elixir contains syntax for that. More, if you want to define your custom type you can do that too. In this lesson we will learn about `@spec` and `@type` syntax.
```

Comment:
Actually this was brought up during OpenCamps.  We should _definitely_ try to move away from comparing Elixir to Ruby and Phoenix to Rails, it does the community a disservice.  I take full blame for that, my original lessons made lots of references.